x-pack/plugins/endpoint/server/routes/resolver.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IRouter } from 'kibana/server';
+import { SearchResponse } from 'elasticsearch';
+import { schema } from '@kbn/config-schema';
+import { EndpointAppContext, Total } from '../types';
+import {
+  getESChildrenQuery,
+  getESNodeQuery,
+  getESChildrenCountQuery,
+  getESNodeCountQuery,
+  PaginationInfo,
+} from '../services/resolver/query_builder';
+import { ResolverSearchHandler } from '../services/resolver/search_handler';
+import { ResolverData } from '../../common/types';
+import { EntityParseError } from '../services/resolver/common';
+
+export function registerResolverRoutes(router: IRouter, endpointAppContext: EndpointAppContext) {
+  const log = endpointAppContext.logFactory.get('resolver-route');
+  const validateQueryObject = schema.object({
+    page_size: schema.maybe(schema.number({ min: 1, max: 1000 })),
+    page_index: schema.maybe(schema.number({ min: 0 })),
+    entity_id: schema.string(),
```

Comment:
Is `entity_id` supposed to be a UUID? Can we extend `schema` to add UUID validation?

Reply:
`entity_id` can have two formats:
`endgame|<endpoint id>|<unique_pid` or some other format that has yet to be defined in the ecs schema. It might be a UUID but I'm not sure. @andrewstucki @webmat 

Once it's defined we could implement a custom schema validator to handle both string formats.

Reply:
@jonathan-buttner Can you double-check that that's true? My code's not using `maybe` and I definitely have tests for the case with no parameters, and those are passing...

Reply:
@jonathan-buttner Nevermind, I think mine's working due to a bug in the schema code. String params require `maybe` ... I'll change mine, I think yours is good. 👍

Reply:
@jonathan-buttner Uggh sorry. You know what it is? If you give a defaultValue, maybe isn't required... so you can either do one or the other, whatever is preferred.

Reply:
Ah ok that makes sense. I think I'll keep mine with the `schema.maybe` and without a `defaultValue` in the handler so that it uses the default value specified in the configuration file so it can be configurable.

Reply:
so the notion of what I'm going to trying to get into ECS core as `entity_id` doesn't necessarily map directly to a UUID, it's more of an arbitrary value that maps to a globally unique  process identifier  on a system -- beats has its own implementation of `entity_id` that we can't really use due to limited entropy -- additionally you can think of things like sysmon-style process guids as filling this role too. In general we should be fine just saying, "it acts as a unique identifier" without fully caring about the implementation details under the hood in these routes.

Reply:
I don't have any context on this PR. All I can say is that we want to get Unique PID https://github.com/elastic/ecs/issues/672 into ECS 1.5, in advance of the next feature freeze.

Quick note: Beats does populate `process.entity_id` (some details in ECS issue above), but it's not an ECS field. It's a custom field populated by Beats.

This is not a problem, but I just wanted to make sure this was clear to everyone concerned :-)